core/desugarEffects.ml

```diff
@@ -95,20 +95,20 @@ let unpack_var_id = function
 
 module SEnv = Env.String
 
-type tycon_info = Kind.t list * bool
+type tycon_info = Kind.t list * bool * Types.typ option
```

Comment:
I think you should add a comment about the purpose of each of the three components.

Reply:
Done!